lxd/storage/backend_lxd.go

```diff
@@ -6239,6 +6265,66 @@ func (b *lxdBackend) BackupCustomVolume(projectName string, volName string, tarW
 	return nil
 }
 
+func (b *lxdBackend) CreateCustomVolumeFromISO(projectName string, volName string, srcData io.ReadSeeker, op *operations.Operation) error {
+	l := b.logger.AddContext(logger.Ctx{"project": projectName, "volume": volName})
+	l.Debug("CreateCustomVolumeFromISO started")
+	defer l.Debug("CreateCustomVolumeFromISO finished")
+
+	// Check whether we are allowed to create volumes.
+	req := api.StorageVolumesPost{
+		Name: volName,
+	}
+
+	err := b.state.DB.Cluster.Transaction(context.TODO(), func(ctx context.Context, tx *db.ClusterTx) error {
+		return project.AllowVolumeCreation(tx, projectName, req)
+	})
+	if err != nil {
+		return fmt.Errorf("Failed checking volume creation allowed: %w", err)
+	}
+
+	revert := revert.New()
+	defer revert.Fail()
+
+	// Get the volume name on storage.
+	volStorageName := project.StorageVolume(projectName, volName)
+
+	vol := b.GetVolume(drivers.VolumeTypeCustom, drivers.ContentTypeISO, volStorageName, nil)
+
+	// Validate config and create database entry for new storage volume.
+	// Strip unsupported config keys (in case the export was made from a different type of storage pool).
+	err = VolumeDBCreate(b, projectName, volName, "", vol.Type(), false, vol.Config(), time.Time{}, time.Time{}, vol.ContentType(), true, true)
+	if err != nil {
+		return fmt.Errorf("Failed creating database entry for custom volume: %w", err)
+	}
+
+	revert.Add(func() { _ = VolumeDBDelete(b, projectName, volName, vol.Type()) })
+
+	_, err = srcData.Seek(0, os.SEEK_SET)
```

Comment:
`.Seek(0, 0)` seems to be what's used elsewhere in the code base. While `staticcheck` recommends replacing `os.SEEK_SET` with `io.SeekStart` maybe we should use `.Seek(0, 0)` here as well.

I couldn't track why seeking to the beginning is needed but my hunch is because of the file type guessing done by reading the first ~32k of the file, is that a right guess?

Reply:
> .Seek(0, 0) seems to be what's used elsewhere in the code base. While staticcheck recommends replacing os.SEEK_SET with io.SeekStart maybe we should use .Seek(0, 0) here as well.

I just read that `os.SEEK_SET` is deprecated. Thanks for the hint. I prefer `io.SeekStart` simply because it's clearer to me where we're seeking.

> I couldn't track why seeking to the beginning is needed but my hunch is because of the file type guessing done by reading the first ~32k of the file, is that a right guess?

Yes, your guess is 100% on point :wink: I need to seek to the beginning otherwise the file will be kinda useless.

Reply:
Indeed, `io.SeekStart` feels less like a magic number ;)